### PR TITLE
A simple RayCast2D example project

### PR DESCRIPTION
**Pull request by WalasPrime**
A simple example project that's supposed to show how RayCast2D works in a nice, graphical way.
The project also encourages use of:

* CanvasItem's `_draw()` and `update()`
* Signals
* Sub-scenes
* AnimationPlayers

### CODE REVIEW

**Comment by TwistedTwigleg** [comment]
Looks good! Everything seems fine and it does exactly what you would expect :+1:
Maybe add a project icon?

**Comment by WalasPrime** [comment]
Gotcha, added a project icon which is basically a small screenshot.

**Comment by MoraleMuffin5** [comment]
@WalasPrime Thanks, looks like it's been a month but $3,000/m is not enough for the lead developers to hit that merge button! Donate some more!

File: 2d/raycast2d/HitObjectManager.gd [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+func _ready():
+	connect("raycast_hit", self, "on_raycast_hit") # Will call on_raycast_hit() when raycast_hit signal is received
+
+func on_raycast_hit():
+	# NOTE: This is called on each frame the raycast hits this node
+	#		Try uncommenting the following lines to prevent this
+	#if get_node("AnimationPlayer").get_current_animation() != "bump" \
+	#or !get_node("AnimationPlayer").is_playing():
+		get_node("AnimationPlayer").play("bump")
```
Comment (aaronfranke) [inline 2d/raycast2d/HitObjectManager.gd]: Please ensure that there are newlines at the end of each file.

File: 2d/raycast2d/RayCastManager.gd [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+var distance_hit = 0
+var hit = false
+
+func _ready():
+	set_enabled(true) # RayCasts are disabled by default
+	set_physics_process(true)
+
+# Process raycast updates every physics frame
+func _physics_process(dt):
```
Comment (aaronfranke) [inline 2d/raycast2d/RayCastManager.gd]: Prefix arguments with `_` to indicate that they are unused.

**Comment by aaronfranke** [comment]
Unfortunately, this PR is just too old, and has not been updated since. I'm also not quite sure if it would be desired to begin with. I'll close this PR, but feel free to discuss things below, or make a new one.

**Pull request closed without merging**